crates/ty_python_semantic/src/types/infer.rs

```diff
@@ -773,7 +774,11 @@ impl<'db> TypeInferenceBuilder<'db> {
                 } else {
                     None
                 }
-            });
+            })
+            .collect();
+
+        // Ensure diagnostics appear in the same order as the class definitions in the source code.
+        class_definitions.sort_unstable_by_key(|(_, node)| node.start());
```

Comment:
Adding snapshots revealed that we weren't reporting all diagnostics in source order if there were multiple `[duplicate-base]` diagnostics in the same file. This fixes that.

Reply:
We should fix this in the mdtest framework. We already sort diagnostics in the CLI

Reply:
I'll revert this for now and accept a snapshot with the diagnostics in an odd order. Will tackle the mdtest sorting in a followup.